### PR TITLE
Chat 52 functions setup

### PR DESCRIPTION
**Pull request by raimondi1337**

### CODE REVIEW

File: setup/index.html [hunk 1]
```diff
@@ -99,12 +99,12 @@
 
           <div class="form-group">
             <label for="email">PubNub Email:</label>
-            <input type="email" class="form-control" id="email" placeholder="Email" value="ian+pubnubgarbage@meetjennings.com">
```
Comment (ianjennings) [inline setup/index.html]: lol forgot about this email address

File: setup/setup.js [hunk 2]
```diff
@@ -1,93 +1,182 @@
...
+                    key.properties.message_storage_ttl = 7;
+                    key.properties.multiplexing = 1;
+                    key.properties.presence_announce_max = 20;
+                    key.properties.presence_debounce = 2;
+                    key.properties.presence_global_here_now = 1;
+                    key.properties.presence_interval = 10;
+                    key.properties.presence_leave_on_disconnect = 0;
+                    key.properties.blocks = 1;
+                    key.properties.uls = 1;
+                    key.properties.wildcardsubscribe = 1;
```
Comment (ianjennings) [inline setup/setup.js]: Oh yeah, these settings need to be reviewed by product. No required changes on your end though.

File: setup/setup.js [hunk 3]
```diff
@@ -1,93 +1,182 @@
...
+                                                status('Done!');
+
+                                                callback(null, { pub: key.publish_key, sub: key.subscribe_key });
+                                            });
+                                        });
+                                    }, 'text');
+                                });
+                            }, 'text');
+                        });
+                    });
```
Comment (ianjennings) [inline setup/setup.js]: async/series can help with the nested callbacks:
https://caolan.github.io/async/docs.html#series

**Comment by ianjennings** [comment]
I'm gonna run this tomorrow with a clean account and test it locally.

File: setup/setup.js (hunk 3 above)
Comment (raimondi1337) [reply setup/setup.js]: I was intending to update to node 8 and using native async/await like I did [here](https://github.com/pubnub/pubnub-open-growth/blob/master/deploy) but didn't feel like figuring out how to plug it into your UI so just continued your pattern

Comment (ianjennings) [reply setup/setup.js]: o(╥﹏╥)o

Comment (ianjennings) [reply setup/setup.js]: I'm so sorry, I know it's all my fault for hacking together too fast

Comment (raimondi1337) [reply setup/setup.js]: Don't be, it works as intended with minimal effort for now. After MVP I'll take a few days and write a nice clean fully featured version.

Comment (ianjennings) [reply setup/setup.js]: I'm just being overdramatic for fun :)

http://icompile.eladkarako.com/wp-content/uploads/2016/01/icompile.eladkarako.com_callback_hell.gif

**Pull request merged**